app/helpers/user_services_helper.rb

```diff
@@ -1,5 +1,84 @@
+# def get_create_user_params
+#   @input = parse_request
+#   print @input
+#   validates_presence_of :email_id, :password, :gender, :address
+# end
+
+def get_create_address_params
+  @input = parse_request
+  validates_presence_of :name, :line1, :line2, :city, :state, :pincode , :phone_number
+end
+
+def create_address(address_params, user_id)
+  address_params[:user_id] = get_valid_user_id user_id
+  address_params[:is_primary] = is_primary_address address_params
+  address = Address.transaction do
+    Address.create!(address_params)
+  end
+  return address
+end
+
+def get_update_address_params
+  @input = parse_request
+  validates_presence_of :address_id, :address
+end
+
+def get_delete_addresses_params
+  @input = parse_request
+  validates_presence_of :address_ids
+end
+
 def get_create_user_params
   @input = parse_request
-  print @input
-  validates_presence_of :email_id, :password, :gender, :address
+  validates_presence_of :email_id
+end
+
+def update_user_params
+  @user_params = {} #defining empty hash
+  @user_params[:email_id] = @input[:email_id]
+  if @input[:password].nil?
+    @user_params[:password] = 'opendoors'
+  end
+
+  if !@input[:gender].nil?
+    @user_params[:gender] = @input[:gender]
+  end
+
+  if !@input[:login_type].nil?
+    @user_params[:login_type] = @input[:login_type]
+  else
+    @user_params[:login_type] = "Google"
+  end
+end
+
+def is_primary_address(address_params)
+  if !address_params.has_key?("is_primary") then
+      return false
+  end
+  return true
+end
+
+def update_address (user_id)
+  existing_address = Address.where('user_id' => user_id, 'id' => @input[:address_id]).as_json;
```

Comment:
Instead of writing new method find_by_id_and_user_id!, I specified as it is.

If i don't use as_json, I couldn't find whether the address is new or old